spring-integration-core/src/main/java/org/springframework/integration/aggregator/DefaultAggregateHeadersFunction.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.aggregator;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.integration.IntegrationMessageHeaderAccessor;
+import org.springframework.integration.store.MessageGroup;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+
+/**
+ * @author Artem Bilan
+ *
+ * @since 5.2
+ */
+public class DefaultAggregateHeadersFunction implements Function<MessageGroup, Map<String, Object>> {
```

Comment:
I wonder if it would be better to introduce `AggregateHeadersFunction extends ...` ?

Reply:
For what reason, Gary?
Essentially we would need to introduce such `Function` extensions everywhere we use `Fucntion`s. It is not done in other Frameworks, like Reactor, so I don't think this is going to be confusing for end-users to figure out that strategy is just `Function<MessageGroup, Map<String, Object>>`.

Well, eventually we might need to think about removing many other functional interface strategies in favor of `java.util.function`...

Reply:
OK